### PR TITLE
33287: Support domain APIs

### PR DESCRIPTION
**Pull request by labkey-nicka**

### CODE REVIEW

File: samples/domain_example.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+labkey_server = 'localhost:8080'
+project_name = 'Study'  # Project folder name
+context_path = 'labkey'
+server_context = create_server_context(labkey_server, project_name, context_path, use_ssl=False)
+
+###################
+# Create a list domain
+###################
+list_domain_definition = {
+    'kind': 'IntList',  # or 'VarList'
```
Comment (labkey-klum) [inline samples/domain_example.py]: Should we have a page on .org at the API level which documents all of the domains which can be created with this api and what the domain kind optional params are.

Comment (labkey-nicka) [reply samples/domain_example.py]: Yes, we should also have API tests (in java?) for all the ones that we state we support.

Comment (labkey-nicka) [reply samples/domain_example.py]: Even better would be an API endpoint that would describe to users the available types and the associated optional parameters.

**Comment by cnathe** [comment]
Nick and I met last week to discuss this and he gave me a tour of the changes in this pull request at that time. I agree that it will be a good reference as we do the work for the Rlabkey package.

**Pull request merged**